pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -326,6 +343,28 @@ func (dswp *desiredStateOfWorldPopulator) markPodProcessed(
 	dswp.pods.processedPods[podName] = true
 }
 
+func (dswp *desiredStateOfWorldPopulator) recordFailedProcessPodEvent(
```

Comment:
Did you see issues with generating too many events?  The events framework should already be aggregating the same events.

Reply:
@msau42 , yeap, the same events are being aggregated now, but based on current volume manager implementation, I can see that there will be more than 5 same events sent to pod per second. As a result, I'm thinking it's better to set up an interval to reduce the number of the same events.